### ISSUE TITLE
OAuth content type

### ISSUE THREAD

**Issue by FalkF** [stablekernel/aqueduct#703]
I was wondering if there is a reason why the AuthContoller does only accepts url encoded login requests. I already found https://tools.ietf.org/html/rfc6749#section-4.1.1 and guess it is taken right from the spec.

Is there a specific reason to not manually allow other content types?

**Comment by joeconwaystk** [comment]
I think its valid to have other content-types (its been a bit since I've read the spec to recall that detail). Since URL encoded form data is supported on every platform (i.e. pure HTML forms without JS), its the catch-all. Cost/time/value is the only reason other content-types aren't implemented today - there isn't a security reason or some such thing AFAIK.

**Comment by FalkF** [comment]
Cool, I just tested this and it is working great after just adding `ContentType.json` to `acceptedContentTypes`.
Are you open for a PR?

**Comment by joeconwaystk** [comment]
Yes, but please add tests to confirm the success case and the case where a required value is missing.

**Comment by joeconwaystk** [comment]
Test file and guidance here: https://github.com/stablekernel/aqueduct/blob/master/aqueduct/test/auth/auth_controller_test.dart
